core/src/fields.rs

```diff
@@ -469,50 +469,150 @@ impl MongoFields {
+                        self.current_field_for_collection = 0;
+                        return true;
+                    }
+                }
+            }
+            if self.dbs.is_empty() {
+                return false;
+            }
+            let db_name = self.dbs.remove(0);
+            let db = mongo_connection.client.database(&db_name);
+            self.current_db_name = db_name;
+            self.collections_for_db = Some(
+                db.list_collections(self.collection_name_filter.clone(), None)
+                    .unwrap(),
+            );
         }
     }
 }
 
 impl MongoStatement for MongoFields {
     // Move the cursor to the next document and update the current row.
     // Return true if moving was successful, false otherwise.
-    fn next(&mut self, _mongo_connection: Option<&MongoConnection>) -> Result<bool> {
-        unimplemented!()
+    fn next(&mut self, mongo_connection: Option<&MongoConnection>) -> Result<bool> {
+        self.current_field_for_collection += 1;
+        Ok(
+            (self.current_field_for_collection as usize) < self.current_col_metadata.len()
+                || self.get_next_metadata(mongo_connection.unwrap()),
+        )
     }
 
     // Get the BSON value for the cell at the given colIndex on the current row.
     // Fails if the first row as not been retrieved (next must be called at least once before getValue).
-    fn get_value(&self, _col_index: u16) -> Result<Option<Bson>> {
-        unimplemented!()
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        // use self.current_col_metadata[current_field_for_collection]
+        // 1 -> TABLE_CAT
+        // 2 -> TABLE_SCHEM  (NULL)
+        // 3 -> TABLE_NAME
+        // 4 -> COLUMN_NAME
+        // 5 -> DATA_TYPE
+        // 6 -> TYPE_NAME
+        // 7 -> COLUMN_SIZE
+        // 8 -> BUFFER_LENGTH
+        // 9 -> DECIMAL_DIGITS
+        // 10 -> NUM_PREC_RADIX
+        // 11 -> NULLABLE
+        // 12 -> REMARKS
+        // 13 -> COLUMN_DEF
+        // 14 -> SQL_DATA_TYPE
+        // 15 -> SQL_DATETIME_SUB
+        // 16 -> CHAR_OCTET_LENGTH
+        // 17 -> ORDINAL_POSITION
+        // 18 -> IS_NULLABLE "YES" or "NO"
+        let get_meta_data = || {
+            self.current_col_metadata
+                .get(self.current_field_for_collection)
+                .unwrap()
+        };
+        Ok(Some(match col_index {
+            1 => Bson::String(self.current_db_name.clone()),
+            2 => Bson::Null,
+            3 => Bson::String(get_meta_data().table_name.clone()),
+            4 => Bson::String(get_meta_data().col_name.clone()),
+            5 => Bson::Int32(get_meta_data().sql_type.0 as i32),
+            6 => Bson::String(get_meta_data().type_name.clone()),
+            7 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            8 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            9 => Bson::Int32(get_meta_data().precision.unwrap_or(0) as i32),
```

Comment:
Ok, so our scale is 0 for everything except dates, lol

Reply:
It's not quite true for DECIMAL/NUMERIC, exact numeric values should return a scale along with the precision.
The issue that even though our values are fixed point numeric, not all of the values in the field have the same scale and precision and also this is not something we get in the schema.
I think as long as we are able to return the correct value for exact numeric, it might be OK. Returning a scale of 0 could confuse app/user into thinking the NUMERIC values are "integers" though. Maybe it would be best to return the max precision and max scale for NUMERIC values?

https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/sql-data-types
> 

SQL_DECIMAL | DECIMAL(p,s) | Signed, exact, numeric value with a precision of at least p and scale s. (The maximum precision is driver-defined.) (1 <= p <= 15; s <= p).[4]
-- | -- | --
SQL_NUMERIC | NUMERIC(p,s) | Signed, exact, numeric value with a precision p and scale s (1 <= p <= 15; s <= p).[4]

Reply:
We don't have exact numeric types in mongodb